File: CalculatorParser.cs
using System;
using System.Globalization;

namespace Calculator
{
    public class CalculatorParser
    {
        public double ParseDisplay(string text)
        {
            if (!double.TryParse(text, NumberStyles.Float, CultureInfo.InvariantCulture, out double value))
            {
                throw new InvalidOperationException("Invalid number in display.");
            }

            return value;
        }
    }
}

File: src/main/java/com/calculator/CalculatorParser.java
package com.calculator;

/**
 * Parses a display string into a numeric value.
 */
public class CalculatorParser {

    public double parseDisplay(String text) {
        try {
            return Double.parseDouble(text);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Invalid number in display.");
        }
    }
}
